test/kaniko_task_test.go

```diff
@@ -89,12 +116,40 @@ func getTask(namespace string, t *testing.T) *v1alpha1.Task {
 					Name:  "kaniko",
 					Image: "gcr.io/kaniko-project/executor",
 					Args: []string{"--dockerfile=/workspace/Dockerfile",
-						fmt.Sprintf("--destination=%s/kanikotasktest", dockerRepo),
+						fmt.Sprintf("--destination=%s", repo),
 					},
 				}},
 			},
 		},
 	}
+	if withSecretConfig {
+		step := task.Spec.BuildSpec.Steps[0]
+		step.VolumeMounts = []corev1.VolumeMount{
+			{
+				Name:      "kaniko-secret",
+				MountPath: "/secrets",
+			},
+		}
+		step.Env = []corev1.EnvVar{
+			{
+				Name:  "GOOGLE_APPLICATION_CREDENTIALS",
+				Value: "/secrets/config.json",
+			},
+		}
+		task.Spec.BuildSpec.Volumes = []corev1.Volume{
+			{
+				Name: "kaniko-secret",
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: "kaniko-secret",
+					},
+				},
+			},
+		}
+		task.Spec.BuildSpec.Steps[0] = step
```

Comment:
do we need this line? I can't quite tell if `step` is a reference to the existing `step`, or if `step := task.Spec.BuildSpec.Steps[0]` makes a copy of it :thinking: